scripts/tidy.sh

```diff
@@ -0,0 +1,60 @@
+#! /bin/bash
+
+# Copyright (C) 2024 Intel Corporation
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+# http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+# 
+# SPDX-License-Identifier: Apache-2.0
+
+
+
+# this is for developer local check only.
+# not used by github action
+
+if [ -z "${MLIR_DIR}" ]; then
+    echo "The environment variable MLIR_DIR is not set."
+    exit 1
+fi
+
+# set PR_REF if your pr target is not main
+PR_REF=${PR_REF:-main}
+
+MERGE_BASE=$(git merge-base $PR_REF HEAD)
+CHANGED_FILES=$(git diff --name-only $MERGE_BASE HEAD | paste -sd' ')
+
+mkdir -p build
+cd build
+
+# if you do not have clang/clang++/clang-tidy
+# please install it
+# conda install -c conda-forge clangxx cxx-compiler clang-tools
+
+cmake ../../ \
+  -DCMAKE_BUILD_TYPE=Release \
+  -DMLIR_DIR=${MLIR_DIR} \
+  -DCMAKE_EXPORT_COMPILE_COMMANDS=True \
+  -DCMAKE_C_COMPILER=$(which clang) \
+  -DCMAKE_CXX_COMPILER=$(which clang++) \
+  -DLLVM_EXTERNAL_LIT=$(which lit)
+
+for f in $(find ./include -name Makefile); do
+  target=$(make -f $f help |grep IncGen); 
+  if [[ $? -eq 0 ]]; then 
+    cd ${f%Makefile} && make ${target#...} && cd -; 
+  fi ;
+done
+
+wget https://raw.githubusercontent.com/llvm/llvm-project/main/clang-tools-extra/clang-tidy/tool/run-clang-tidy.py -O run-clang-tidy.py
```

Comment:
shall we cache result?

Reply:
@Menooker 
I'm not sure it is ok to place code from llvm in our public repository

Reply:
@kurapov-peter @ZhennanQin What do you think if we copy `.clang-tidy` to our repo?

Reply:
Or do not download if already exists:
```shell
[ -f run-clang-tidy.py ] || wget ...
[ -f .clang-tidy ] || wget ... 
```

Reply:
> @kurapov-peter @ZhennanQin What do you think if we copy `.clang-tidy` to our repo?

sure, @leshikus please take a look